go-controller/pkg/ovn/gateway_cleanup.go

```diff
@@ -94,8 +94,6 @@ func (oc *Controller) gatewayCleanup(nodeName string) error {
 		return fmt.Errorf("failed to delete external switch %s, error: %v", exGWexternalSwitch, err)
 	}
 
-	// We don't know the gateway mode as this is running in the master, try to delete the additional local
```

Comment:
same as what I mentioned in the below comment, - at this point in time we would have only the 1004 policy for packets destined to the host in both the modes. Since the upgrade path takes care of deleting these lgw specific policies and that is going to be a one time operation we should be good. Of course if users used an unsupported upgrade path like an old old lgw version and skipped multiple topology versions or something then we'd be in trouble.

Reply:
I should update the comment though since I removed the no op for non-gateway modes part.